skglm/penalties/separable.py

```diff
@@ -8,18 +8,21 @@
     ST, box_proj, prox_05, prox_2_3, prox_SCAD, value_SCAD, prox_MCP, value_MCP)
 
 
-spec_L1 = [
-    ('alpha', float64),
-]
-
-
-@jitclass(spec_L1)
 class L1(BasePenalty):
     """L1 penalty."""
 
     def __init__(self, alpha):
         self.alpha = alpha
 
+    def get_spec(self):
+        spec_L1 = (
+            ('alpha', float64),
+        )
+        return spec_L1
+
+    def params_to_dict(self):
```

Comment:
is this method to align with scikit learn API?

Reply:
we need it because we get the instance, compile its class (not the instance itself), then need to reinstantiate one instance of the compiled class with the same parameters as the original object